bookkeeper-common/src/main/java/org/apache/bookkeeper/common/component/LifecycleListener.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.bookkeeper.common.component;
+
+/**
+ * Listener listening of the lifecycle changes.
+ */
+public interface LifecycleListener {
+
+    void beforeStart();
```

Comment:
you can consider adding 'default' method implementations, it will make simpler to implement the interface and extend in the guture

Reply:
I don't have a strong preference here. if you feel strong about this, I can change.

Reply:
As we are introducing now the interface there is no real need. We will do it maybe in future.
Ok for me